pkg/controller/job/job_controller.go

```diff
@@ -1757,3 +1768,17 @@ func countReadyPods(pods []*v1.Pod) int32 {
 	}
 	return cnt
 }
+
+// This checks if we should apply PodRecreationPolicy.
```

Comment:
@mimowo lost your comment but I added two functions to consolidate all the feature checks.  

We have a case where we want to count terminating pods and one where we want to apply failed.

Reply:
The checks are backwards, with respecet to PodFailurePolicy. See comments below.

Reply:
updated.